### ISSUE TITLE
Operation with native integer constants that might overflow should be a warning only

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#42955]
The compiler should report a warning rather than an error for `c2` and calculate the value at runtime rather than compile-time.
```C#
class Program
{
    static void Main()
    {
        const nint m = int.MaxValue;
        const nint c1 = checked(m + 1); // error: overflows at compile time
              nint c2 = checked(m + 1); // warning: may overflow
    }
}
```
See https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-03-25.md#behavior-of-constant-folding
